gateway/validate.py

```diff
@@ -0,0 +1,55 @@
+import subprocess
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+class OCFLFixityValidator(ABC):
+    @abstractmethod
+    def validate_repository(self, no_fixity: bool = False, log_level: Optional[str] = None, suppress_warning: Optional[str] = None) -> str:
+        pass
+
+    @abstractmethod
+    def validate_object(self, object_id: str, no_fixity: bool = False, log_level: Optional[str] = None, suppress_warning: Optional[str] = None) -> str:
+        pass
+
+    @abstractmethod
+    def validate_multiple_objects(self, object_paths: List[str], no_fixity: bool = False, log_level: Optional[str] = None, suppress_warning: Optional[str] = None) -> str:
+        pass
+
+    @abstractmethod
+    def _run_rocfl_command(self, command: List[str]) -> str:
+        pass
```

Comment:
I think I'd leave this out of the ABC, since it's `_` (intended to be private). And other implementations likely aren't using `rocfl`.